### PR TITLE
B2CA-1681: Port Flex

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
This PR update the Apps to prepare and then add Flex support:

- Update version to `1.5.0`
- Fix deprecated warnings
- Port to `Standard App` Makefile
- Port to `lib_standard_app`
- Replace `THROW` by `return`
- Add linters config files
- Fix linter comments
- Port _NBGLv2_, and adapt tests accordingly
- Update **Stax** snapshots for _NBGLv2_
- Add **Flex** support

# Checklist

- [X] App update process has been followed 
- [X] Target branch is `develop` 
- [X] Application version has been bumped

### CODE REVIEW

File: src/eos_stream.c [hunk 1]
```diff
@@ -204,8 +205,8 @@ static void processEosioUnlinkAuth(txProcessingContext_t *context) {
 }
 
 static void processUnknownAction(txProcessingContext_t *context) {
-    cx_hash(&context->dataSha256->header, CX_LAST, context->dataChecksum, 0,
-            context->dataChecksum, sizeof(context->dataChecksum));
+    LEDGER_ASSERT(cx_hash_no_throw(&context->dataSha256->header, CX_LAST, context->dataChecksum, 0,
```
Comment (XavierChapron) [inline src/eos_stream.c]: Couldn't it be done using `CX_ASSERT()`?

**Comment by XavierChapron** [comment]
Also changes from here could be interesting  https://github.com/LedgerHQ/app-eos/pull/25/files

**Comment by cedelavergne-ledger** [comment]
> Also changes from here could be interesting https://github.com/LedgerHQ/app-eos/pull/25/files

Yes, I have based my modifications on this PR, taking into accout more recent NBGL APIs :wink:

File: README.md [hunk 2]
```diff
@@ -4,92 +4,204 @@ Eos wallet application framework for Ledger devices
 
 This follows the specification available in the doc/ folder
```
Comment (tdejoigny-ledger) [inline README.md:5]: add flex each time there is also stax in this file

File: src/ui_nbgl.c [hunk 3]
```diff
@@ -32,143 +32,93 @@
...
+    settingContents.contentsList = contents;
+    settingContents.nbContents = 1;
+
+    infosList.nbInfos = 1;
+    infosList.infoTypes = INFO_TYPES;
+    infosList.infoContents = INFO_CONTENTS;
+
+    nbgl_useCaseHomeAndSettings(APPNAME,
+                                &C_app_eos_64px,
+                                "This app confirms actions on\nthe EOS network",
```
Comment (tdejoigny-ledger) [inline src/ui_nbgl.c]: replace `"This app confirms actions on\nthe EOS network"` by NULL

File: tests/functional/usage.md [hunk 4]
```diff
@@ -16,9 +15,10 @@ sudo apt-get update && sudo apt-get install qemu-user-static
...
 The application to test must be compiled for all required devices.
 You can use for this the container `ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite`:
-```
+
+```shell
 docker pull ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite:latest
-cd <your app repository>                        # replace <appname> with the name of your app, (eg boilerplate)
+cd <your app repository>
 docker run --user "$(id -u)":"$(id -g)" --rm -ti -v "$(realpath .):/app" --privileged -v "/dev/bus/usb:/dev/bus/usb" ledger-app-builder-lite:latest
 make clean && make BOLOS_SDK=$<device>_SDK      # replace <device> with one of [NANOS, NANOX, NANOSP, STAX]
```
Comment (tdejoigny-ledger) [inline tests/functional/usage.md]: add flex

File: tests/functional/usage.md [hunk 5]
```diff
@@ -27,47 +27,53 @@ exit
...
 The application to test must be loaded and started on a Ledger device plugged in USB.
 You can use for this the container `ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite`:
-```
+
+```shell
 docker pull ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite:latest
-cd app-<appname>/                                   # replace <appname> with the name of your app, (eg boilerplate)
+cd app-<appname>
 docker run --user "$(id -u)":"$(id -g)" --rm -ti -v "$(realpath .):/app" --privileged -v "/dev/bus/usb:/dev/bus/usb" ledger-app-builder-lite:latest
 make clean && make BOLOS_SDK=$<device>_SDK load     # replace <device> with one of [NANOS, NANOX, NANOSP, STAX]
```
Comment (tdejoigny-ledger) [inline tests/functional/usage.md]: add flex

File: tests/functional/usage.md [hunk 6]
```diff
@@ -27,47 +27,53 @@ exit
...
     -k <testname>   only run the tests that contain <testname> in their names
     --tb=short      in case of errors, formats the test traceback in a readable way
-``` 
+```
 
 Custom pytest options
-```
+
+```shell
     --device <device>           run the test on the specified device [nanos,nanox,nanosp,stax,all]. This parameter is mandatory
```
Comment (tdejoigny-ledger) [inline tests/functional/usage.md]: add flex

**Review by tdejoigny-ledger — COMMENTED** [review]
good practice for next time. Open a dedicated PR for linting, it will ease the delta review.

File: tests/functional/usage.md (hunk 6 above)
Comment (cedelavergne-ledger) [reply tests/functional/usage.md]: Done

File: tests/functional/usage.md (hunk 5 above)
Comment (cedelavergne-ledger) [reply tests/functional/usage.md]: Done

File: tests/functional/usage.md (hunk 4 above)
Comment (cedelavergne-ledger) [reply tests/functional/usage.md]: Done

File: src/ui_nbgl.c (hunk 3 above)
Comment (cedelavergne-ledger) [reply src/ui_nbgl.c]: Done.
Just note this was the old string :wink:

File: README.md (hunk 2 above)
Comment (cedelavergne-ledger) [reply README.md:5]: Done

**Review by tdejoigny-ledger — APPROVED** [review]

**Pull request merged**